users/create.go

```diff
@@ -94,13 +97,42 @@ func init() {
 type createUserRequestBody struct {
 	Email    string `json:"email"    req:"nonzero"`
 	Password string `json:"password" req:"nonzero"`
+	AwsToken string `json:"awsToken"`
+}
+
+//checkAwsTokenLegitimacy check if the AWS Token exist. It return the product code and
+//the customer identifier. If there is no Token, this function is not call.
+func checkAwsTokenLegitimacy(token string) (*marketplacemetering.ResolveCustomerOutput, error) {
+	var awsInput marketplacemetering.ResolveCustomerInput
+	mySession := session.Must(session.NewSession(&aws.Config{
+		Region: aws.String(config.AwsRegion),
+	}))
+	svc := marketplacemetering.New(mySession)
+	awsInput.SetRegistrationToken(token)
+	result, err := svc.ResolveCustomer(&awsInput)
+	if err != nil {
+		fmt.Print(err.Error())
+	}
+	return result, err
 }
 
 func createUser(request *http.Request, a routes.Arguments) (int, interface{}) {
 	var body createUserRequestBody
 	routes.MustRequestBody(a, &body)
 	tx := a[db.Transaction].(*sql.Tx)
-	code, resp := createUserWithValidBody(request, body, tx)
+	var result *marketplacemetering.ResolveCustomerOutput
+	var err error
+	//Check legitimacy of the AWS Token and get User Registration Token
+	var awsCustomerConvert = ""
+	if body.AwsToken != "" {
+		result, err = checkAwsTokenLegitimacy(body.AwsToken)
+		if err != nil {
+			return 409, errors.New("AWS Token error")
```

Comment:
Be slightly more accurate, `Failed to check the AWS token.`
(not too much, it's an error the user can see)